packages/web-components/@sfx/sayt/test/sayt.test.ts

```diff
@@ -65,11 +72,244 @@ describe('Sayt Component', () => {
+      const showSayt = stub(sayt, 'showSayt');
+      const isCorrectSayt = stub(sayt, 'isCorrectSayt').returns(false);
+      const event = {};
+
+      sayt.showCorrectSayt(event);
+
+      expect(isCorrectSayt).to.be.calledOnceWith(event);
+      expect(showSayt).to.not.be.called;
+    });
+  });
+
+  describe('hideCorrectSayt()', () => {
+    it('should call hideSayt() when event specifies correct searchbar ID ', () => {
+      const hideSayt = stub(sayt, 'hideSayt');
+      const isCorrectSayt = stub(sayt, 'isCorrectSayt').returns(true);
+      const event = {};
+
+      sayt.hideCorrectSayt(event);
+
+      expect(isCorrectSayt).to.be.calledOnceWith(event);
+      expect(hideSayt).to.be.calledOnce;
+    });
+
+    it('should not call hideSayt() when event specifies wrong searchbar ID', () => {
+      const hideSayt = stub(sayt, 'hideSayt');
+      const isCorrectSayt = stub(sayt, 'isCorrectSayt').returns(false);
+      const event = {};
+
+      sayt.hideCorrectSayt(event);
+
+      expect(isCorrectSayt).to.be.calledOnceWith(event);
+      expect(hideSayt).to.not.be.called;
+    });
+  });
+
+  describe('isCorrectSayt()', () => {
+    it('should return true if event provides the correct searchbar ID', () => {
+      const searchbar = sayt.searchbar = 'some-searchbar-id';
+      const event = { detail: { searchbar } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return true if event does not provide a searchbar ID', () => {
+      const searchbar = sayt.searchbar = 'some-searchbar-id';
+      const event = { detail: {} };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should return false if event provides the wrong searchbar ID', () => {
+      sayt.searchbar = 'correct-searchbar-id';
+      const event = { detail: { searchbar: 'wrong-searchbar-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.false;
+    });
+
+    it('should return true if event specifies a searchbar but SAYT has none specified', () => {
+      sayt.searchbar = undefined;
+      const event = { detail: { searchbar: 'some-searchbar-id' } };
+
+      const result = sayt.isCorrectSayt(event);
+
+      expect(result).to.be.true;
+    });
+
+    it('should not error if passed an event without a detail attribute', () => {
+      const event = {};
+      const callback = () => sayt.isCorrectSayt(event);
+
+      expect(callback).to.not.throw();
+    });
+  });
+
   describe('render()', () => {
     it('should return an instance of TemplateResult', () => {
       const result = sayt.render();
 
       expect(result).to.be.an.instanceof(TemplateResult);
     });
   });
+
+  describe('processClick()', () => {
+    let node: any = 'some-node';
+    let event: Event;
+
+    beforeEach(() => {
+      event = {
+        target: node,
+      } as Event;
+      sayt.nodeInSearchBar = () => false;
```

Comment:
I think we've been using stubs for these.
```suggestion
      stub(sayt, 'nodeInSearchBar').returns(false);
```

Reply:
I will make this change because I think an assertion could be made that `.nodeInSearchBar()` is actually called (which I will add). In the current state of the code, though, there is no usage of that functionality. This was an intentional choice.